feat: document Actor run modes for scoped tokens (#1310)
apify/apify-docs: sources/platform/integrations/programming/api.md

**A scoped token can access only those resources that you'll explicitly allow it to.**

:::info

> We currently do not allow scoped tokens to create or modify Actors. If you do need to create or modify Actors via Apify API, use an unscoped token.

Reviewer: I'm really not a fan of saying "currently" cause the reader has no way of knowing what is "currently" unless they take a dive into git commit history. 
```suggestion
Scoped tokens cannot create or modify Actors. To create or modify Actors through the Apify API, use an unscoped token.
```

Author: Fair enough :+1:  Dropped "currently" and slightly modified the wording, taking inspiration from you.

Author: Fair enough 👍 Dropped "currently" and slightly modified the wording, taking inspiration from you.

When you run an Actor with a scoped token in this mode, Apify will inject an _unscoped_ token to the Actor runtime.

This way you can be sure that once you give a token the permission to run an Actor, it will just work, and you don't have to worry about the exact permissions the Actor might need. However, this also means that you need to trust the Actor.

:::tip
Use this mode if you want to integrate with a 3rd-party service to run your Actors. Create a scoped token that can only run the Actor you need, and share it with the service. Even if the token is leaked, it can't be used to access your other data.
:::

#### Restricted access: Restrict what Actors can access using the scope of this Actor

When you run an Actor with a scoped token in this mode, Apify will inject a token with the same scope as the scope of the original token.

This way you can be sure that Actors won't accidentally—or intentionally—access any data they shouldn't. However, Actors might not function properly if the scope is not sufficient.

:::tip
This restriction is _transitive_, which means that if the Actor runs another Actor, its access will be restricted as well.
:::

#### Default run storages

> When Apify runs an Actor, it automatically creates a set of default storages (a dataset, a key-value store and request queue) that the Actor can use in runtime.

Reviewer: Maybe link here to each part of documentation?

Author: Good idea, added.

Reviewer: Oh I thought more to dataset, request queue and Key-value store ( or maybe just one link under storages?)

Reviewer: Fair enough, it works either way
